src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointResponse.cs

```diff
@@ -21,47 +18,35 @@ internal class EndpointResponse
     public bool IsIResult { get; set; }
     public bool IsSerializable { get; set; }
     public bool IsEndpointMetadataProvider { get; set; }
-    private WellKnownTypes WellKnownTypes { get; init; }
 
-    internal EndpointResponse(IMethodSymbol method, WellKnownTypes wellKnownTypes)
+    internal EndpointResponse(IMethodSymbol method)
     {
-        WellKnownTypes = wellKnownTypes;
         ResponseType = UnwrapResponseType(method, out bool isAwaitable, out bool awaitableIsVoid);
         WrappedResponseType = method.ReturnType.ToDisplayString(EmitterConstants.DisplayFormat);
         IsAwaitable = isAwaitable;
         HasNoResponse = method.ReturnsVoid || awaitableIsVoid;
         IsIResult = GetIsIResult();
         IsSerializable = GetIsSerializable();
         ContentType = GetContentType();
-        IsEndpointMetadataProvider = ImplementsIEndpointMetadataProvider(ResponseType, wellKnownTypes);
+        IsEndpointMetadataProvider = ImplementsIEndpointMetadataProvider(ResponseType);
     }
 
-    private static bool ImplementsIEndpointMetadataProvider(ITypeSymbol? responseType, WellKnownTypes wellKnownTypes)
-        => responseType == null ? false : responseType.Implements(wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_Metadata_IEndpointMetadataProvider));
+    private static bool ImplementsIEndpointMetadataProvider(ITypeSymbol? responseType)
+        => responseType == null ? false : responseType.Implements(["Microsoft", "AspNetCore", "Http", "Metadata", "IEndpointMetadataProvider"]);
```

Comment:
I'm not familiar with the new initialization syntax. Does this array get cached?